internal/clients/authentication.go

```diff
@@ -0,0 +1,65 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+)
+
+type Authentication interface {
+	// Is checks if Authentication is of a given provider type
+	Is(providerType models.ProviderType) bool
+
+	// MustBe returns nil, if authentication is of given time. Otherwise, returns an error.
+	MustBe(providerType models.ProviderType) error
+
+	// String returns authentication string (ARN, Subscription UUID, Project-ID...)
+	String() string
+}
+
+type auth struct {
+	ProviderType models.ProviderType `json:"type"`
+	Payload      string              `json:"payload"`
```

Comment:
I  am not sure payload is the best name here

Reply:
I am not sure either, but recommend a better alternative. It's either ARN or subscription id or project id.